### ISSUE TITLE
Default sorting of flags ect to not be created time stamp

### ISSUE THREAD

**Issue by alucarddelta** [flipt-io/flipt#3413]
### Problem

Is it possible to change the default sorting of the flag table to be something other than the created timestamp.

When trying to view a bunch of flags, it is much easier to look for a key/name in alphabetical order.

### Ideal Solution

Have a config option under the UI section for default sort order, Allowing config values of:
- key
- name
- type
- created

eg
```yaml
 ui:
     table_sorting:
          flag: key
          segments: name
```

### Search

- [X] I searched for other open and closed issues before opening this

**Comment by markphelps** [comment]
Great idea @alucarddelta ! What if we saved the latest sort field/direction in local storage? That would prevent the need for setting it in config

**Comment by alucarddelta** [comment]
> Great idea @alucarddelta ! What if we saved the latest sort field/direction in local storage? That would prevent the need for setting it in config

That also works :)

**Comment by markphelps** [comment]
@alucarddelta I prototyped a fix here where instead of using local storage I just save the sorting state for the table to Redux

see: https://github.com/flipt-io/flipt/pull/3415

This means that if you close the browser it will clear the preferred saved sorting field/direction.. but it will no longer lose the sorting field/direction if you remain in the app and click around to different pages.

In this gif, I first show the current behavior, that whenever you click away from the flags table it resets the sorting state

Then I switch to the new behavior that uses Redux to store the flag table state, which keeps it persisted between pages. 

Would this solve your issue?

![2024-08-28 19 48 24](https://github.com/user-attachments/assets/f155f776-49f8-4ea1-9861-292b23843775)

**Comment by alucarddelta** [comment]
That will def be helpful!... This solution would work well when setting up new applications/namespace with lots of back and forth setting up flags, rules, ect. I am slightly concerned when accessing Flipt UI infrequently between sessions to make minor changes and toggling flags manually.

As part of this FR/Idea, I did have a thought last night came from one of our internal tools... Rather than using storage to set default sort, we chose to uses URL query parameters to store table options. This includes things like pagination, limits, search filters, and Sort Order. Is this something that would be useful in your mind?

eg.
```/foo?FoosFilter-limit=100&FoosFilter-%5B%5DorderBy=foo_name```

**Comment by markphelps** [comment]
> That will def be helpful!... This solution would work well when setting up new applications/namespace with lots of back and forth setting up flags, rules, ect. I am slightly concerned when accessing Flipt UI infrequently between sessions to make minor changes and toggling flags manually. But that shouldn't be a massive issue and can be dealt with.
> 
> As part of this FR/Idea, I did have a thought last night came from one of our internal tools... Rather than using storage to set default sort, we chose to uses URL query parameters to store table options. This includes things like pagination, limits, search filters, and Sort Order. Is this something that would be useful in your mind?
> 
> eg. `/foo?FoosFilter-limit=100&FoosFilter-%5B%5DorderBy=foo_name`

I think we could support that as well. The UI is a Single Page Application (SPA) but we can get query params from the URL and still make the JS handle pagination/sorting/etc. I will see if we can support this easily

**Comment by erka** [comment]
resolved by #3415
